### ISSUE TITLE
Quick question about batched inputs in MIL

### ISSUE THREAD

**Issue by fcoelhomrc** [mahmoodlab/MIL-Lab#13]
Hey, awesome work! Nice to see these repos from MahmoodLab promoting standardization! 

I have a question regarding batching in your MIL implementations / training pipelines.

How do you batch inputs when bag tensors have an inhomogeneous dimension (variable number of patches)? 

I am often stuck with unit batch size in my projects because of this. Is there anything I am missing?

**Comment by danielshaowers** [comment]
Glad to hear you're finding the work useful! For batching with inhomogeneous shapes, I typically just use a unit batch size as well. However, you could also subsample a fixed number of patches (2048, 4096) in your training dataset's '__getitem__', as is done in THREADS, to obtain a larger batch size. This subsampling also acts as a form of regularization which may improve generalizability by preventing the model from relying on specific patches. You could also consider padding followed by attention masking, or gradient accumulation to simulate larger batch sizes.

Hope that's helpful!
